core/src/main/java/org/mifos/mobilewallet/core/domain/model/Transaction.java

```diff
@@ -161,4 +161,142 @@ public String toString() {
+            if(pos==1) {
+                if(input.length()>= 2) {
+                    String temp=input.substring(input.length()-2,input.length());
+                    input=input.substring(0,input.length()-2);
+                    converted+=digits(temp);
+                } else if(input.length()==1) {
+                    converted+=digits(input);
+                    input="";
+            }
+                pos++;
+            } else if(pos==2) {
+                String temp=input.substring(input.length()-1,input.length());
+                input=input.substring(0,input.length()-1);
+                if(converted.length()> 0&&digits(temp)!="") {
+                    converted=(digits(temp)+maxs[pos]+" and")+converted;
+                    hun=true;
+                } else {
+                    if(digits(temp)=="") {
+                    } else {
+                        converted = (digits(temp) + maxs[pos]) + converted;
+                        hun = true;
+                    }
+                }
+                pos++;
+            } else if(pos > 2) {
+                if(input.length()>= 2) {
+                    String temp=input.substring(input.length()-2,input.length());
+                    input=input.substring(0,input.length()-2);
+                    if(!hun&&converted.length()> 0) {
+                        converted = digits(temp) + maxs[pos] + " and" + converted;
+                    } else {
+                        if(digits(temp)=="")  {
+                        } else {
+                            converted = digits(temp) + maxs[pos] + converted;
+                        }
+                    }
+                } else if(input.length()==1) {
+                    if(!hun&&converted.length()> 0) {
+                        converted = digits(input) + maxs[pos] + " and" + converted;
+                    } else {
+                        if(digits(input)=="") {
+                        } else {
+                            converted = digits(input) + maxs[pos] + converted;
+                        }
+                        input="";
+                    }
+                }
+                pos++;
+            }
+        }
+        return converted+" Only";
+    }
+    // TO RETURN SELECTED NUMBERS IN WORDS
+    private String digits(String temp) {
+        String converted="";
+        for(int i=temp.length()-1;i >= 0;i--) {
+            int ch=temp.charAt(i)-48;
+            if(i==0&&ch>1 && temp.length()> 1) {
+                converted = tens[ch - 2] + converted;
+            } else if(i==0&&ch==1&&temp.length()==2) {
+                int sum=0;
+                for(int j=0;j < 2;j++)
+                    sum=(sum*10)+(temp.charAt(j)-48);
+                return teen[sum-10];
+            } else {
+                if(ch > 0)
+                    converted=units[ch]+converted;
+            }
+        }
+        return converted;
+    }
+    // CONVERT THE NUMBER TO STRING
+    private String numToString(int x) {
```

Comment:
There is no need to create this function you can simply use pre-build java function `Integer.toString(int)`